recipe/deactivate.bat

```diff
@@ -0,0 +1,23 @@
+@echo off
+:: Ensure delayed variable expansion
+setlocal enabledelayedexpansion
+
+:: Initialize a new variable to store the cleaned path
+set "NEW_CMAKE_PREFIX_PATH="
```

Comment:
Instead you set CMAKE_PREFIX_PATH_BACKUP=CMAKE_PREFIX_PATH at the beginning of the activation script.
Then in the deactivate script CMAKE_PREFIX_PATH=CMAKE_PREFIX_PATH_BACKUP.

I think that's how it's done usually. For example: https://github.com/AnacondaRecipes/cuda-nvcc-feedstock/blob/main/recipe/deactivate.bat

Reply:
As discussed offline, I added a comment explaining the rationale. Let me know if that makes sense or not.